src/backend/distributed/master/master_node_protocol.c

```diff
@@ -648,7 +648,7 @@ GetTableDDLEvents(Oid relationId)
 		{
 			isConstraint = true;
 		}
-		else if (indexForm->indisunique)
+		else if (indexForm->indisunique || indexForm->indisexclusion)
```

Comment:
I think this change is not sufficient. We also need to add shardId to the exclude constraint index on the workers to prevent creating the same named exclusion indexes when replication factor > 2. See the example below:
```SQL

CREATE TABLE exclusion_const_overlaps_operator
(
	partition_col tsrange,
	other_col tsrange,
	CONSTRAINT exclusion_const_overlaps_operator_excl EXCLUDE USING gist (other_col WITH &&, partition_col WITH =)
);

SELECT master_create_distributed_table('exclusion_const_overlaps_operator', 'partition_col', 'hash');

SELECT master_create_worker_shards('exclusion_const_overlaps_operator', '2', '2');

WARNING:  relation "exclusion_const_overlaps_operator_excl" already exists
CONTEXT:  while executing command on localhost:9701
WARNING:  could not create shard on "localhost:9701"
ERROR:  could only create 1 of 2 of required shard replicas
```
Which probably ties into `RelayEventExtendNames()` function.

(We should also consider dropping the table as we consider creating shards)